### PR TITLE
[Settings][Screensaver] Add setting to disable screensaver while playing audio

### LINKED ISSUE #22956: Dims during music visualization
### Describe the bug
Here is a clear and concise description of what the problem is:

For at least a year or two, through numerous releases, during music visualization the screen will dim after the screensaver timeout.  Doesn't matter what skin, or even that I have Slideshow set as my screensaver (which works when music is not playing).

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

When the Settings/Interface/Screensaver "Use Visualization When Playing Audio" switch is on, the screen should not dim during music play.  This worked properly for many years.

## Actual Behavior

Screen dims after 1 minute (my screensaver selected timeout period)

## Possible Fix

Have the Settings switch work again.

### To Reproduce

1.  Set screensaver timeout to 1 minute
2. Play a song and activate music visualization (tab key)
3. Wait one minute for screen to dim

### Debuglog

The debuglog can be found here:

https://pastebin.com/raw/YApdaiTT

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] macOS
 - [x ] Windows 7x64
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version:
kodi-20.0-Nexus-x64

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by enen92**
## Description
After a while investigating https://github.com/xbmc/xbmc/issues/22956 I think I've finally found what the reported issue is. Currently we have an option to disable the screensaver (e.g. dim) if a visualisation add-on is set and we are playing audio -> this works well. However, I doubt most users will realise that the artist slideshow or any image displaying and animating in the background of the music OSD does not count as a visualisation. For such cases, the screensaver will be triggered.
Hence this PR adds a generic setting to disable the screensaver being activated while playing audio and also reorganises the settings a bit better on the screensaver section (splitting between audio and video). Also disables the visualization setting if no visualization addon is set (to reduce confusion amongst users).

@howie-f for review (it's just a couple of ifs), @KarellenX will need wiki if goes in due to the new setting (and improved screen)

## Motivation and context
Closes https://github.com/xbmc/xbmc/issues/22956

## How has this been tested?
Playing audio with the setting enabled and no visualization set, confirming screensaver does not dim. Checking if a visualization is enabled the visualization triggers and is kept activated (without dim) just like before.

## What is the effect on users?
Should now be able to avoid the screensaver if playing audio, globally.
Settings should also be a bit more organised now.

## Screenshots (if appropriate):
image

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

File: addons/resource.language.en_gb/resources/strings.po [hunk 1]
```diff
@@ -7082,10 +7083,7 @@ msgctxt "#13391"
 msgid "Eject / Load"
 msgstr ""
 
-#: system/settings/settings.xml
-msgctxt "#13392"
-msgid "Use visualisation if playing audio"
-msgstr ""
+#empty strings with id 13392
```
Comment (howie-f) [inline addons/resource.language.en_gb/resources/strings.po:7086]:
```suggestion
#empty string with id 13392
```

File: addons/resource.language.en_gb/resources/strings.po [hunk 2]
```diff
@@ -19185,11 +19183,7 @@ msgctxt "#36132"
 msgid "Preview the selected screensaver."
 msgstr ""
 
-#. Description of setting with label #13392 "Use visualisation if playing audio"
-#: system/settings/settings.xml
-msgctxt "#36133"
-msgid "If music is being played, the selected visualisation will be started instead of displaying the screensaver."
-msgstr ""
+#empty strings with id 36133
```
Comment (howie-f) [inline addons/resource.language.en_gb/resources/strings.po:19186]:
```suggestion
#empty string with id 36133
```

File: addons/resource.language.en_gb/resources/strings.po [hunk 3]
```diff
@@ -23434,3 +23428,14 @@ msgstr ""
 msgctxt "#39191"
 msgid "HDR"
 msgstr ""
+
+#: system/settings/settings.xml
+msgctxt "#39192"
+msgid "Disable screensaver if playing audio"
```
Comment (howie-f) [inline addons/resource.language.en_gb/resources/strings.po:23434]: hmmmm. `if` or `when` ?
```suggestion
msgid "Disable screensaver when playing audio"
```

**Review by howie-f — APPROVED** [review]
looks good imo. minor suggestions. thank you

**Review by howie-f — APPROVED** [review]

**Comment by enen92** [comment]
jenkins build and merge

**Comment by 3dfx** [comment]
I don't think the old setting was redundant, it was indeed, very neat.
This PR introduces some misbehavior that may or may not be the way it was intented.

When I'm in front of the TV on the couch, I use the visualisation in fullscreen.
And while working, not facing the TV, I turn on the screensaver via an RPC command.

I have set my screensaver to "black screen".
If I am playing any music now, it doesn't, after the configured screensaver timeout, switch to visualisation and stays in the dialog you are in (instead of turning visualisation into fullscreen).
This is for OLED tv owners very bad as you stay for an undefined amount of time with static menus (risk of burn in).

And if I switch to the new mode, I switch to visualisation and after 1 minute (configured screensaver timeout), it activates my screensaver....... why?
I just actively switch to the visualization.

So, as a workaround I have to call via RPC to use the screensaver:
curl -H Content-Type:application/json -d '{"jsonrpc": "2.0", "method": "Settings.SetSettingValue", "params":{"setting":"screensaver.disableforaudio", "value":'false'},"id": 1}'

And to use the visualisation as I want to be in fullscreen, with disable = true.

**Pull request merged**